apps/api/src/server/applications/s51advice/s51-advice.validators.js

```diff
@@ -40,3 +41,67 @@ export const validatePaginationCriteria = composeMiddleware(
 		.optional({ nullable: true }),
 	validationErrorHandler
 );
+
+export const validateS51AdviceToUpdateProvided = composeMiddleware(
+	body('[]').notEmpty().withMessage('Must provide S51 Advice to update'),
+	validationErrorHandler
+);
+
+/**
+ * Validate that an array of S51 Advice ids exist
+ *
+ * @param {{id: number }[]} s51Adviceids
+ */
+const validateAllS51AdviceExists = async (s51Adviceids) => {
+	if (s51Adviceids) {
+		for (const adviceId of s51Adviceids) {
+			try {
+				const adviceRecord = await s51AdviceRepository.get(adviceId.id);
+
+				if (adviceRecord === null || typeof adviceRecord === 'undefined') {
+					throw new Error(`Unknown S51 Advice id ${adviceId.id}`);
+				}
+			} catch {
+				throw new Error(`Unknown S51 Advice id ${adviceId.id}`);
+			}
```

Comment:
Could be useful to pass along the original error too.

```suggestion
			} catch (err) {
				throw new Error(`Unknown S51 Advice id ${adviceId.id}: ${err}`);
			}
```

Reply:
I see your point, though in this case though the error is mostly the one thrown in the try, and then it dbl outputs the message - eg "Unknown S51 Advice id 400: Error: Unknown S51 Advice id 400"